packages/ra-ui-materialui/src/list/SingleFieldList.js

```diff
@@ -54,68 +54,67 @@ const sanitizeRestProps = ({
  *     </SingleFieldList>
  * </ReferenceManyField>
  */
-export class SingleFieldList extends Component {
+function SingleFieldList({
+    classes: classesOverride,
+    className,
+    ids,
+    data,
+    loaded,
+    resource,
+    basePath,
+    children,
+    linkType,
+    ...rest
+}) {
+    const classes = useStyles({ classes: classesOverride });
+
     // Our handleClick does nothing as we wrap the children inside a Link but it is
     // required fo ChipField which uses a Chip from material-ui.
     // The material-ui Chip requires an onClick handler to behave like a clickable element
-    handleClick = () => {};
-    render() {
-        const {
-            classes = {},
-            className,
-            ids,
-            data,
-            loaded,
-            resource,
-            basePath,
-            children,
-            linkType,
-            ...rest
-        } = this.props;
+    const handleClick = () => {};
```

Comment:
move this function out of the component to avoid passing a new function at every render

Reply:
done